### PR TITLE
Grafana v8.5

### LINKED ISSUE #90: Unable to send using grafana 8.5
Pretty much as the title says - some of the structures have changed in v8.5

### PR DESCRIPTION
**Pull request by mark-prins**
Closes #90 

I'm wondering about leaving this on a separate branch and not merging in. Thoughts?

### CODE REVIEW

File: pkg/datasource/report_content.go [hunk 1]
```diff
@@ -27,8 +27,8 @@ func (datasource *MsupplyEresDatasource) GetReportContent(scheduleID string) ([]
 	defer rows.Close()
 
 	for rows.Next() {
-		var ID, ScheduleID, DashboardID, Variables, Lookback string
-		var PanelID int
+		var ID, ScheduleID, DashboardID, Variables string
+		var PanelID, Lookback int
```
Comment (Mai21) [inline pkg/datasource/report_content.go:31]: @mark-prins The reason why @sworup changed Lookback to string is mentioned in this [comment](https://github.com/openmsupply/excel-report-email-scheduler/issues/76#issuecomment-1161315333)
Could you please have a look at this?

Comment (mark-prins) [reply pkg/datasource/report_content.go:31]: thanks @Mai21 - I don't understand what the problem is from that comment.. if you could help explain? 🤷 
The reason that I changed it from a string is that the grafana structure expects an int, and passing a string results in problems running the report. I spent some time adding in conversions but the end result is changing to an int. that's working for me on v8.5

Comment (Mai21) [reply pkg/datasource/report_content.go:31]: Hmm, I see. I am wondering if Grafana v8.5 fixed the issue about `$__timeTo()` 

> Just did some testing and found that grafana expects epoch timestamp in microseconds, while the date conversion in our code at least is converting in seconds. Now epoch timestamp system could be seconds, miliseconds or microseonds but $__timeTo() method does not do the check and expect the time to be milliseconds hence putting in the oldest 1970s date time... this ends up getting blank values.

As my understanding, according to Sworup's investigation, `$__timeTo()` method used to expect `milliseconds` while our code was converting in seconds. And String From "now - 5m",  To "now" worked as we expected, that's why he changed it to String. And in PNG, Grafana v8.5 & this Lookback string version of the plugin are working...

But anyway if the int works properly and Grafana expects an int, there is no way not to go with int I suppose.

**Comment by sworup** [comment]
@mark-prins I had found that in our report-mailer plugin if we passing lookback as epoch timestamp would not work. It would send email and all but the data would be blank or incorrect (not within the lookback period).

It has to do with epoch timestamp millisecond and second discrepancy and grafana does to convert it and add correct date range in the query. 

So I used human readable string which the plugin excepted and was working. Found it strange that it did not work for you.

It is strange that the change you have made worked as I have also changed the dropdown fields which are still in human readable string form:

https://github.com/openmsupply/excel-report-email-scheduler/blob/550239704588e1215733d120116329cdd474caa8/src/constants.ts#L44-L54

**Comment by sworup** [comment]
@mark-prins I suggest we do the Sqlite database field changes for lookback fields and not use sting data.

**Comment by mark-prins** [comment]
I think this has been superseded now by your changes @sworup - will just close this one

**Pull request closed without merging**